web3.js/src/connection.ts

```diff
@@ -2557,31 +2558,60 @@ export class Connection {
     configOrCommitment?: GetMultipleAccountsConfig | Commitment,
   ): Promise<(AccountInfo<Buffer | ParsedAccountData> | null)[]> {
     const keys = publicKeys.map(key => key.toBase58());
+    const chunks = sliceIntoChunks(keys, 100);
 
-    let commitment;
-    let encoding: 'base64' | 'jsonParsed' = 'base64';
-    if (configOrCommitment) {
-      if (typeof configOrCommitment === 'string') {
-        commitment = configOrCommitment;
-        encoding = 'base64';
-      } else {
-        commitment = configOrCommitment.commitment;
-        encoding = configOrCommitment.encoding || 'base64';
+    let accounts : (AccountInfo<Buffer | ParsedAccountData> | null)[] = [];
+    let count = 0;
+    let maxData = 0;
+
+    for (let i = 0; i < chunks.length; i++) {
+
+      // this enforces a 10 second pause periodically to ensure rate limits aren't exceeded
+      if (count === 25) {
+        await sleep(10000);
```

Comment:
Explicit sleeps like this should be avoided.  The RPC servers will respond with 429 with a [Retry-After](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) header indicating the client must throttle

Reply:
Thanks for the feedback. I'll remove the explicit sleep calls and re-submit. Still new to the Solana ecosystem, so I appreciate this being pointed out!

Reply:
Removed the sleep calls and adjusted the size of chunks. Testing locally everything works. Looking forward to your feedback.